### PR TITLE
WalletConnect QR scan fixes and add retry UI

### PR DESCRIPTION
**Pull request by MananTank**
## PR-Codex overview
This PR enhances WalletConnect connection handling by displaying Retry option on failure/rejection.

### Detailed summary
- Added error handling and Retry option in WalletConnect connection UI
- Updated chain selection logic for WalletConnect initialization
- Imported `AccentFailIcon` for error display in `ScanScreen`
- Added error handling props and Retry function in `ScanScreen`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 1]
```diff
@@ -220,6 +221,7 @@ async function initProvider(
     optionalMethods: OPTIONAL_METHODS,
     optionalEvents: OPTIONAL_EVENTS,
     optionalChains: chainsToRequest,
+    chains: chainsToRequest.length > 0 ? [chainsToRequest[0]] : [1],
```
Comment (MananTank) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:224]: following the same pattern as v4 here

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 2]
```diff
@@ -499,6 +509,13 @@ function getChainsToRequest(options: {
       ? (optionalChainIds as ArrayOneOrMore<number>)
       : [1];
 
+  if (!options.chain && optionalChains.length === 0) {
+    rpcMap[1] = getRpcUrlForChain({
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:513]: should use getCachedChain here i think otherwise it wont respect the users RPC override

Comment (MananTank) [reply packages/thirdweb/src/wallets/wallet-connect/index.ts]: done

Comment (MananTank) [reply packages/thirdweb/src/wallets/wallet-connect/index.ts]: done

**Review by gregfromstl — APPROVED** [review]

**Pull request merged**